### PR TITLE
PIN-3691 When create a purpose a draft version will be created at the same time

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/purposemanagement/api/impl/PurposeApiServiceImpl.scala [hunk 1]
```diff
@@ -60,7 +60,17 @@ final case class PurposeApiServiceImpl(
...
+    val purposeVersion: PersistentPurposeVersion =
+      PersistentPurposeVersion.fromPurposeSeed(purposeSeed, uuidSupplier, dateTimeSupplier)
+
+    val result: Future[Purpose] = for {
+      newPurpose <- commander(purpose.id.toString)
+        .askWithStatus(CreatePurpose(purpose, _))
+        .map(_.toAPI)
+      newVersion <- commander(purpose.id.toString)
+        .askWithStatus(CreatePurposeVersion(purpose.id.toString, purposeVersion, _))
+        .map(_.toAPI)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposemanagement/api/impl/PurposeApiServiceImpl.scala]: these should be replaced with a single call.
The `CreatePurpose` takes a full purpose, in which you can attach the version

**Review by galales — CHANGES_REQUESTED** [review]

File: src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionSpec.scala [hunk 2]
```diff
@@ -20,73 +19,76 @@ class PurposeVersionSpec extends BaseIntegrationSpec {
...
-        freeOfChargeReason = None
+        freeOfChargeReason = None,
+        dailyCalls = 100
       )
-      val versionSeed = PurposeVersionSeed(dailyCalls = 100)
-
-      val response: Future[PurposeVersion] =
-        for {
-          _      <- createPurpose(purposeId, purposeSeed)
-          result <- createPurposeVersion(purposeId, versionId, versionSeed)
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionSpec.scala:37]: this is testing the creation of a single new version, it should not be changed to the creation of the purpose + first version

File: src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionSpec.scala [hunk 3]
```diff
@@ -20,73 +19,76 @@ class PurposeVersionSpec extends BaseIntegrationSpec {
...
+        freeOfChargeReason = None,
+        dailyCalls = 100
       )
-      val riskAnalysisDoc = PurposeVersionDocument(
-        id = riskAnalysisId,
-        contentType = "a-content-type",
-        path = "a/store/path",
-        createdAt = timestamp
-      )
-      val versionSeed     = PurposeVersionSeed(riskAnalysis = Some(riskAnalysisDoc), dailyCalls = 100)
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionSpec.scala:74]: same, this is testing the creation of a version with the risk analysis document. It should be kept

File: src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala [hunk 4]
```diff
@@ -10,47 +10,6 @@ import scala.concurrent.Future
 class PurposeVersionStateChangeSpec extends BaseIntegrationSpec {
 
   "Activation of purpose" must {
-    "succeed when risk analysis is added at creation" in {
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala]: any reason why this has been deleted?
We should still be able to load risk analysis at creation and then activate

File: src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala [hunk 5]
```diff
@@ -73,25 +32,24 @@ class PurposeVersionStateChangeSpec extends BaseIntegrationSpec {
...
         for {
-          _       <- createPurpose(purposeId, purposeSeed)
-          version <- createPurposeVersion(purposeId, versionId, versionSeed)
-          result  <- activateVersion(purposeId, versionId, ChangedBy.CONSUMER, Some(riskAnalysisDoc))
-        } yield (version, result)
+          _      <- createPurpose(purposeId, versionId, purposeSeed)
+          result <- activateVersion(purposeId, versionId, ChangedBy.CONSUMER, Some(riskAnalysisDoc))
+        } yield result
 
-      val (version, result) = response.futureValue
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala:87]: this is testing that on activation, those (and only) specific fields are updated.
This is why it was returning the created version and the activated version.
With this update, if a new field that we do not expect is changed the test will succeed.

This should be rollbacked

File: src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala [hunk 6]
```diff
@@ -373,26 +331,25 @@ class PurposeVersionStateChangeSpec extends BaseIntegrationSpec {
...
-          _ <- activateVersion(purposeId, versionId, ChangedBy.CONSUMER, versionSeed.riskAnalysis, firstActivationAt)
-          _ <- suspendVersion(purposeId, versionId, ChangedBy.CONSUMER)
+          _      <- createPurpose(purposeId, versionId, purposeSeed)
+          _      <- activateVersion(purposeId, versionId, ChangedBy.CONSUMER, Some(riskAnalysisDoc), firstActivationAt)
+          _      <- suspendVersion(purposeId, versionId, ChangedBy.CONSUMER)
           result <- activateVersion(purposeId, versionId, ChangedBy.CONSUMER, None, secondActivationAt)
-        } yield (version, result)
+        } yield result
 
-      val (version, result) = response.futureValue
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala:389]: same for rollback

File: src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala [hunk 7]
```diff
@@ -515,25 +470,24 @@ class PurposeVersionStateChangeSpec extends BaseIntegrationSpec {
...
-      val response: Future[(PurposeVersion, PurposeVersion)] =
+      val response: Future[PurposeVersion] =
         for {
-          _       <- createPurpose(purposeId, purposeSeed)
-          version <- createPurposeVersion(purposeId, versionId, versionSeed)
-          _       <- activateVersion(purposeId, versionId, ChangedBy.CONSUMER, versionSeed.riskAnalysis)
-          result  <- suspendVersion(purposeId, versionId, ChangedBy.CONSUMER)
-        } yield (version, result)
-
-      val (version, result) = response.futureValue
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala:530]: same for rollback

File: src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala [hunk 8]
```diff
@@ -607,25 +560,24 @@ class PurposeVersionStateChangeSpec extends BaseIntegrationSpec {
...
-      val response: Future[(PurposeVersion, PurposeVersion)] =
+      val response: Future[PurposeVersion] =
         for {
-          _       <- createPurpose(purposeId, purposeSeed)
-          version <- createPurposeVersion(purposeId, versionId, versionSeed)
-          _       <- activateVersion(purposeId, versionId, ChangedBy.CONSUMER, versionSeed.riskAnalysis)
-          result  <- archiveVersion(purposeId, versionId, ChangedBy.CONSUMER)
-        } yield (version, result)
-
-      val (version, result) = response.futureValue
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala:622]: same for rollback

File: src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala [hunk 9]
```diff
@@ -691,24 +642,23 @@ class PurposeVersionStateChangeSpec extends BaseIntegrationSpec {
...
         for {
-          _       <- createPurpose(purposeId, purposeSeed)
-          version <- createPurposeVersion(purposeId, versionId, versionSeed)
-          result  <- waitForApprovalVersion(purposeId, versionId, ChangedBy.CONSUMER)
-        } yield (version, result)
+          _      <- createPurpose(purposeId, versionId, purposeSeed)
+          result <- waitForApprovalVersion(purposeId, versionId, ChangedBy.CONSUMER)
+        } yield result
 
-      val (version, result) = response.futureValue
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposemanagement/provider/PurposeVersionStateChangeSpec.scala:705]: same for rollback

File: src/main/scala/it/pagopa/interop/purposemanagement/model/persistence/Adapters.scala [hunk 10]
```diff
@@ -193,7 +193,19 @@ object Adapters {
       id = uuidSupplier.get(),
       eserviceId = seed.eserviceId,
       consumerId = seed.consumerId,
-      versions = Seq.empty,
+      versions = Seq(
+        PersistentPurposeVersion(
+          id = uuidSupplier.get(),
+          state = Draft,
+          dailyCalls = seed.dailyCalls,
+          createdAt = dateTimeSupplier.get(),
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposemanagement/model/persistence/Adapters.scala]: this is subtle but very useful when troubleshooting:
may you create the timestamp with `dateTimeSupplier.get()` just once and use the same value in both purpose and version?

**Review by galales — APPROVED** [review]

**Pull request merged**